sdk/Lusid.Sdk/Utilities/ApiRetryHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Polly;
+using RestSharp;
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Class used to define API error retry rules for all API calls.
+    /// </summary>
+    public static class ApiRetryHandler
+    {
+        /// <summary>
+        /// Number of max retry attempts
+        /// </summary>
+        public const int MaxRetryAttempts = 3;
+
+        /// <summary>
+        /// Get the internal exception condition on which to retry.
+        /// </summary>
+        /// <param name="restResponse">Response object that comes from the API Client</param>
+        /// <returns></returns>
+        private static bool GetInternalExceptionRetryCondition(IRestResponse restResponse)
+        {
+            return restResponse.ErrorException != null || restResponse.StatusCode == 0;
```

Comment:
Why/when would the statusCode be `0`? In what circumstances does the `restResponse.ErrorException` exist? Will this fire on all non 2xx status codes?

Reply:
It's not clear to me at all when a retry will occur and when it won't.

Reply:
Status code is 0 whenever for example the http connection is abruptly aborted, or a timeout happens, or any other exception that originates from within the SDK.

When the above happens, an ErrorException object will also be added to the response

It shouldn't trigger on any other status code than 0, unless these staus codes throw an exception internally somewhere

Reply:
What other name would you give to `GetInternalExceptionRetryCondition` to make it more self explanatory?

Reply:
Or maybe it just needs updating the comment?

Reply:
Added plenty of examples in the latest MR

Reply:
Okay great, nice